### PR TITLE
Replace api to get all boundaries ids with api to get all boundaries infos (id, filename, scenario time)

### PR DESCRIPTION
**Pull request by FranckLecuyer**
PR that must be merged before (migration merge orchestrator to Postgres) : 
https://github.com/gridsuite/merge-orchestrator-server/pull/50
https://github.com/gridsuite/deployment/pull/145

All following PR must be merged together : 
https://github.com/gridsuite/cgmes-boundary-server/pull/15
https://github.com/gridsuite/cgmes-boundary-import-job/pull/3
https://github.com/gridsuite/merge-orchestrator-server/pull/51
https://github.com/gridsuite/merge-notification-server/pull/16
https://github.com/gridsuite/gridmerge-app/pull/67

### CODE REVIEW

File: src/test/java/org/gridsuite/boundary/importer/job/BoundaryImportJobTest.java [hunk 1]
```diff
@@ -172,11 +174,20 @@ public void mainTest() throws IOException {
 
         // all boundaries already present in cgmes boundary server, no boundary will be imported
         mockServer.getClient().clear(request());
-        addGetBoundariesIdsExpectation("[\"urn:uuid:22222222-bbbb-bbbb-bbbb-bbbbbbbbbbbb\", \"urn:uuid:11111111-aaaa-aaaa-aaaa-aaaaaaaaaaaa\"]");
+        addGetBoundariesInfosExpectation("[{\"id\":\"urn:uuid:22222222-bbbb-bbbb-bbbb-bbbbbbbbbbbb\",\"filename\":\"20210315T0000Z__ENTSOE_TPBD_002.xml\",\"scenarioTime\":\"2020-02-02T00:00\"},{\"id\":\"urn:uuid:11111111-aaaa-aaaa-aaaa-aaaaaaaaaaaa\",\"filename\":\"20210315T0000Z__ENTSOE_EQBD_002.xml\",\"scenarioTime\":\"2020-02-02T18:35\"}]");
```
Comment (SlimaneAmar) [inline src/test/java/org/gridsuite/boundary/importer/job/BoundaryImportJobTest.java:177]: Maybe a matcher ?

File: src/main/java/org/gridsuite/boundary/importer/job/CgmesBoundaryServiceRequester.java [hunk 2]
```diff
@@ -92,28 +93,29 @@ public boolean importBoundary(TransferableFile boundaryFile) throws IOException,
...
-                List<String> result = new ArrayList<>();
+                List<BoundaryInfo> result = new ArrayList<>();
                 JSONArray array = new JSONArray(json);
                 for (int i = 0; i < array.length(); i++) {
-                    result.add(array.getString(i));
+                    var obj = array.getJSONObject(i);
+                    result.add(new BoundaryInfo(obj.getString("id"), obj.getString("filename"), LocalDateTime.parse(obj.getString("scenarioTime"))));
                 }
                 return result;
             }
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/boundary/importer/job/CgmesBoundaryServiceRequester.java:114]: Log an error massage if status != 200
            } else {
                LOGGER.error(response.toString());
            }

File: src/test/java/org/gridsuite/boundary/importer/job/BoundaryImportJobTest.java [hunk 3]
```diff
@@ -163,7 +164,8 @@ public void mainTest() throws IOException {
 
         // 1 boundary already present in cgmes boundary server, only 1 boundary will be imported
         mockServer.getClient().clear(request());
-        addGetBoundariesIdsExpectation("[\"urn:uuid:22222222-bbbb-bbbb-bbbb-bbbbbbbbbbbb\"]");
+        addGetBoundariesInfosExpectation("[{\"id\":\"urn:uuid:22222222-bbbb-bbbb-bbbb-bbbbbbbbbbbb\",\"filename\":\"20210315T0000Z__ENTSOE_TPBD_002.xml\",\"scenarioTime\":\"2020-02-02T00:00\"}]");
```
Comment (SlimaneAmar) [inline src/test/java/org/gridsuite/boundary/importer/job/BoundaryImportJobTest.java:167]: Maybe a matcher ?

File: src/main/java/org/gridsuite/boundary/importer/job/CgmesBoundaryServiceRequester.java [hunk 4]
```diff
@@ -92,28 +93,29 @@ public boolean importBoundary(TransferableFile boundaryFile) throws IOException,
...
         } catch (IOException e) {
-            LOGGER.error("I/O Error while getting all boundary ids");
+            LOGGER.error("I/O Error while getting all boundary infos");
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/boundary/importer/job/CgmesBoundaryServiceRequester.java:116]: Log the cause ?
e.printStackTrace();

File: src/main/java/org/gridsuite/boundary/importer/job/CgmesBoundaryServiceRequester.java [hunk 5]
```diff
@@ -92,28 +93,29 @@ public boolean importBoundary(TransferableFile boundaryFile) throws IOException,
...
             if (response.statusCode() == 200) {
                 String json = response.body();
 
-                List<String> result = new ArrayList<>();
+                List<BoundaryInfo> result = new ArrayList<>();
                 JSONArray array = new JSONArray(json);
                 for (int i = 0; i < array.length(); i++) {
-                    result.add(array.getString(i));
+                    var obj = array.getJSONObject(i);
+                    result.add(new BoundaryInfo(obj.getString("id"), obj.getString("filename"), LocalDateTime.parse(obj.getString("scenarioTime"))));
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/boundary/importer/job/CgmesBoundaryServiceRequester.java:111]: Why do not user an object mapper ?
            if (response.statusCode() == 200) {
                ObjectMapper mapper = new ObjectMapper();
                mapper.registerModule(new JavaTimeModule());
                return mapper.readValue(response.body(), new TypeReference<List<BoundaryInfo>>(){});
            } else {
                LOGGER.error(response.toString());
            }

Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/boundary/importer/job/CgmesBoundaryServiceRequester.java:111]: Done

File: src/main/java/org/gridsuite/boundary/importer/job/CgmesBoundaryServiceRequester.java (hunk 4 above)
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/boundary/importer/job/CgmesBoundaryServiceRequester.java:116]: Done

File: src/main/java/org/gridsuite/boundary/importer/job/CgmesBoundaryServiceRequester.java (hunk 2 above)
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/boundary/importer/job/CgmesBoundaryServiceRequester.java:114]: Done

**Review by SlimaneAmar — APPROVED** [review]

**Review by niconoir — APPROVED** [review]

**Pull request merged**